query/src/main/scala/filodb/query/exec/rangefn/InstantFunction.scala

```diff
@@ -362,18 +364,33 @@ final case class HistogramQuantileImpl() extends HistToDoubleIFunction {
   }
 }
 
+/**
+ * Histogram quantile function for Histogram columns, where all buckets are together. This will take in consideration
+ * of min and max columns
+ */
+final case class HistogramQuantileWithMaxMinImpl() extends HMaxMinToDoubleIFunction {
+  final def apply(value: Histogram, max: Double, min: Double, scalarParams: Seq[Double]): Double = {
+    require(scalarParams.length == 1, "Quantile (between 0 and 1) required for histogram quantile")
+    val maxMinHist = value match {
+      case h: MutableHistogram => MaxMinHistogram(h, max, min)
+      case other: Histogram => MaxMinHistogram(MutableHistogram(other), max, min)
+    }
+    maxMinHist.quantile(scalarParams(0))
+  }
+}
+
 /**
  * Histogram max quantile function for Histogram column + extra max (Double) column.
  */
-final case class HistogramMaxQuantileImpl() extends HDToDoubleIFunction {
+final case class HistogramMaxQuantileImpl() extends HMaxMinToDoubleIFunction {
   /**
     * @param scalarParams - a single value between 0 and 1, the quantile to calculate.
     */
-  final def apply(hist: Histogram, max: Double, scalarParams: Seq[Double]): Double = {
+  final def apply(hist: Histogram, max: Double, min: Double = Double.NaN, scalarParams: Seq[Double]): Double = {
     require(scalarParams.length == 1, "Quantile (between 0 and 1) required for histogram quantile")
     val maxHist = hist match {
-      case h: MutableHistogram => MaxHistogram(h, max)
-      case other: Histogram    => MaxHistogram(MutableHistogram(other), max)
+      case h: MutableHistogram => MaxMinHistogram(h, max)
+      case other: Histogram    => MaxMinHistogram(MutableHistogram(other), max)
```

Comment:
Should `min` be passed as an argument here, too, if it's not `Double.NaN`?

Reply:
+1, good catch @alextheimer. i think in this case, since `HistogramMaxQuantileImpl` is invoked only when historgram schema as only `max` column and not `min`. may be make `min` as default 0 instead of NaN and pass it along?

Reply:
Need default for min as  Double.NaN for the HistMaxMinAggregator ( if not, then it forcibly brings down the min to 0 even if the min is set to a higher number like 10 or 20 )

let me pass 0 for HistogramMaxQuantile function